src/NewTwitchApi/Resources/AbstractResource.php

```diff
@@ -36,20 +36,82 @@ protected function callApi(string $uriEndpoint, array $queryParamsMap = [], stri
      * $queryParamsMap should be a mapping of the param key expected in the API call URL,
      * and the value to be sent for that key.
      *
-     * [['key' => 'param_key', 'value' => 42],['key' => 'other_key', 'value' => 'asdf']]
-     * would result in
-     * ?param_key=42&other_key=asdf
+     * For example, the following array:
+     *
+     * [
+     *   'int_key'    => 42,
+     *   'bool_key'   => true,
+     *   'string_key' => 'twitch',
+     *   'array_key'  => ['is', 'number', 1],
+     * ]
+     *
+     * would result in the following query params:
+     *
+     *  ?int_key=42&bool_key=true&string_key=twitch&array_key=is&array_key=number&array_key=1
      */
     protected function generateQueryParams(array $queryParamsMap): string
     {
         $queryStringParams = '';
-        foreach ($queryParamsMap as $paramMap) {
-            if ($paramMap['value']) {
-                $format = is_int($paramMap['value']) ? '%d' : '%s';
-                $queryStringParams .= sprintf('&%s='.$format, $paramMap['key'], $paramMap['value']);
+
+        foreach ($queryParamsMap as $key => $value) {
+            $queryStringParams .= $this->generateQueryStringFragment($key, $value);
+        }
+
+        return $queryStringParams ? '?' . substr($queryStringParams, 1) : '';
+    }
+
+    /**
+     * @param mixed $value
+     */
+    private function generateQueryStringFragment(string $key, $value): string
+    {
+        switch (gettype($value)) {
+            case 'string':
+                return $this->generateStringFragment($key, $value);
+            case 'integer':
+                return $this->generateIntegerFragment($key, $value);
+            case 'boolean':
+                return $this->generateBooleanFragment($key, $value);
+            case 'array':
+                return $this->generateArrayFragement($key, $value);
+            default:
+                return '';
```

Comment:
I think this should throw an exception instead of returning an empty string. We only expect/allow the four types explicitly handled in the `case` statements. If something else is used, the issue should be visible with an exception. The way this is now, it will just fail silently.